agent/provider/src/market/negotiator/builtin/allow_only.rs

```diff
@@ -0,0 +1,49 @@
+use crate::market::negotiator::{NegotiationResult, NegotiatorComponent};
+use crate::provider_agent::AgentNegotiatorsConfig;
+use crate::rules::restrict::RuleValidator;
+use crate::rules::{CheckRulesResult, RulesManager};
+
+use ya_agreement_utils::ProposalView;
+use ya_manifest_utils::DEMAND_MANIFEST_NODE_DESCRIPTOR_PROPERTY;
+
+pub struct AllowOnly {
+    rules: RulesManager,
+}
+
+impl AllowOnly {
+    pub fn new(agent_negotiators_cfg: AgentNegotiatorsConfig) -> Self {
+        Self {
+            rules: agent_negotiators_cfg.rules_manager,
+        }
+    }
+}
+
+impl NegotiatorComponent for AllowOnly {
+    fn negotiate_step(
+        &mut self,
+        demand: &ProposalView,
+        offer: ProposalView,
+    ) -> anyhow::Result<NegotiationResult> {
+        let node_descriptor = demand
+            .get_property::<serde_json::Value>(DEMAND_MANIFEST_NODE_DESCRIPTOR_PROPERTY)
+            .ok();
+
+        match self
+            .rules
+            .allow_only()
+            .check_rule(demand.issuer, node_descriptor)
+        {
+            CheckRulesResult::Accept => Ok(NegotiationResult::Ready { offer }),
+            CheckRulesResult::Reject(reason) => {
+                log::debug!(
+                    "[AllowOnly] Rejecting Proposal from Requestor {}, reason: {reason}",
+                    demand.issuer
+                );
+                Ok(NegotiationResult::Reject {
+                    message: reason,
+                    is_final: true,
+                })
+            }
+        }
```

Comment:
`allow-only` does not check `blacklist` (and `blacklist` does not check `allow-only`).
Composite negotiator stores negotiators in HashMap and these are iterated in some non arbitrary order.
I created following `allow-only` and `blacklist`:
```
"blacklist": {
  "enabled": true,
  "identity": [
    "0x9acce8db6bdbcae218be2ecf15175ca3c5f227fa"
  ],
  "certified": [
    "cefd51ab9b18986ad11870f2f0e54418d8bbf425b36f30420027bcf3f3d2e11e8533b939272308ede90f57df9242a0d5463f07fdb2199417a4b2c56c9ac00114"
  ]
},
"allow-only": {
  "enabled": true,
  "identity": [
    "0x9acce8db6bdbcae218be2ecf15175ca3c5f227fa"
  ],
  "certified": [
    "cefd51ab9b18986ad11870f2f0e54418d8bbf425b36f30420027bcf3f3d2e11e8533b939272308ede90f57df9242a0d5463f07fdb2199417a4b2c56c9ac00114"
  ]
}
```
Will the negotiation result depend only on negotiators order?